packages/stitch/tests/nestedRootTypes.test.ts

```diff
@@ -0,0 +1,78 @@
+import { makeExecutableSchema } from "@graphql-tools/schema";
+import { stitchSchemas } from "@graphql-tools/stitch";
+import { graphql } from 'graphql';
+
+describe('nested root types', () => {
+  const schema1 = makeExecutableSchema({
+    typeDefs: `
+      type Query {
+        schema1Boolean: Boolean
+        schema1Query: Query
+      }
+    `,
+    resolvers: {
+      Query: {
+        schema1Boolean: () => true,
+        schema1Query: () => ({}),
+      },
+    },
+  });
+
+  const schema2 = makeExecutableSchema({
+    typeDefs: `
+      type Query {
+        schema2Boolean: Boolean
+        schema2Query: Query
```

Comment:
I didn’t even know that was a thing..,

Reply:
It is a thing! Usually used in the context of mutations, as mutations are done serially, but sometimes you might want to follow-up a mutation with a query, and you can't mix mutations and queries in the same root operation, and the ordering is only defined with mutations, but by nesting a query root type within a mutation response, you can do something similar, see: https://graphql-rules.com/rules/mutation-payload-query